packages/auth/src/providers/cognito/utils/signInHelpers.ts

```diff
@@ -32,6 +36,153 @@ import {
 } from '../../../types';
 import { AuthError } from '../../../errors/AuthError';
 import { InitiateAuthException } from '../types/errors';
+import {
+	AllowedMFATypes,
+	AuthUserAttribute,
+	TOTPSetupDetails,
+} from '../../../types/models';
+import { verifySoftwareTokenClient } from './clients/VerifySoftwareTokenClient';
+
+import { associateSoftwareTokenClient } from './clients/AssociateSoftwareTokenClient';
+import { AuthErrorCodes } from '../../../common/AuthErrorStrings';
+import { AuthValidationErrorCode } from '../../../errors/types/validation';
+import { assertValidationError } from '../../../errors/utils/assertValidationError';
+import { signInStore } from './signInStore';
+
+const USER_ATTRIBUTES = 'userAttributes.';
+
+export async function handleCustomChallenge(
+	challengeResponse: string,
+	clientMetadata: ClientMetadata | undefined,
+	session: string | undefined,
+	username: string
+): Promise<RespondToAuthChallengeCommandOutput> {
+	const challengeResponses = { USERNAME: username, ANSWER: challengeResponse };
+	const jsonReq: RespondToAuthChallengeClientInput = {
+		ChallengeName: 'CUSTOM_CHALLENGE',
+		ChallengeResponses: challengeResponses,
+		Session: session,
+		ClientMetadata: clientMetadata,
+	};
+	return await respondToAuthChallengeClient(jsonReq);
+}
+
+export async function handleMFASetupChallenge(
+	challengeResponse: string,
+	clientMetadata: ClientMetadata | undefined,
+	session: string | undefined,
+	username: string,
+	deviceName?: string
+): Promise<RespondToAuthChallengeCommandOutput> {
+	const challengeResponses = {
+		USERNAME: username,
+	};
+
+	const { Session } = await verifySoftwareTokenClient({
+		UserCode: challengeResponse,
+		Session: session,
+		FriendlyDeviceName: deviceName,
+	});
+
+	signInStore.dispatch({
+		type: 'SET_ACTIVE_SIGN_IN_SESSION',
+		value: Session,
+	});
+
+	const jsonReq: RespondToAuthChallengeClientInput = {
+		ChallengeName: 'MFA_SETUP',
+		ChallengeResponses: challengeResponses,
+		Session,
+		ClientMetadata: clientMetadata,
+	};
+	return await respondToAuthChallengeClient(jsonReq);
```

Comment:
Nit (can just return the promise)

```suggestion

	return respondToAuthChallengeClient(jsonReq);
```